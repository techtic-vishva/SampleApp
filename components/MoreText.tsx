import React, { useCallback, useState } from 'react'
import { View } from 'react-native'
import { chattanoogaTapWater } from '../constants/Colors'
import { Text } from './Themed'

export default function ({ text }: { text?: string }) {
	const [isReadMore, setReadMore] = useState(false)
	const [lengthMore, setLengthMore] = useState(false)
	const onTextLayout = useCallback((e) => {
		setLengthMore(e.nativeEvent.lines.length >= 4)
	}, [])
	return (
		<>
			{text !== null && text !== '' ? (
				<View>
					<Text
						onTextLayout={onTextLayout}
						numberOfLines={isReadMore ? undefined : 4}
						ellipsizeMode="tail"
						style={{
							color: chattanoogaTapWater,
							fontSize: 14,
							marginTop: 7
						}}>
						{text}
					</Text>
					{lengthMore && !isReadMore && (
						<Text
							onPress={() => setReadMore(true)}
							style={{
								position: 'absolute',
								backgroundColor: 'rgba(0, 0, 0, 0.9)',
								paddingStart: 5,
								paddingVertical: 1,
								bottom: 0,
								right: 0,
								color: 'white',
								fontSize: 14,
								alignSelf: 'flex-end'
							}}>
							more
						</Text>
					)}
				</View>
			) : null}
		</>
	)
}
