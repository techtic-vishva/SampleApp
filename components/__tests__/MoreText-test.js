import * as React from 'react'
import renderer from 'react-test-renderer'
import MoreText from '../MoreText'

it('render when passing the props', () => {
	const tree = renderer.create(<MoreText text={''} />).toJSON()
	expect(tree).toMatchSnapshot()
})
