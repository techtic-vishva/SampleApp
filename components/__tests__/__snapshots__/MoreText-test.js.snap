// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when passing the props 1`] = `null`;
